src/query/v2/plan/operator.cpp

```diff
@@ -1233,6 +1233,55 @@ class AggregateCursor : public Cursor {
     return true;
   }
 
+  void PullMultiple(MultiFrame &multi_frame, ExecutionContext &context) override {
+    SCOPED_PROFILE_OP("AggregateMF");
+
+    if (!pulled_all_input_) {
+      while (!pulled_all_input_) {
+        ProcessAll(multi_frame, &context);
+      }
+      multi_frame.MakeAllFramesInvalid();
+      aggregation_it_ = aggregation_.begin();
+
+      // in case there is no input and no group_bys we need to return true
+      // just this once
```

Comment:
This comment seems false

Reply:
If you check the first condition (`if (!pulled_all_input_)`), that will be true only on the first run, so here we only return true only one.